src/OIGenerator.cpp

```diff
@@ -76,8 +76,20 @@ OIGenerator::findOilTypesAndNames(drgnplusplus::program& prog) {
       linkageName = linkageNameCstr;
     }
 
+    const std::string_view linkagePrefix =
+        "_ZN19ObjectIntrospection13getObjectSize";
```

Comment:
This isn't a stable name - so I'm not sure if we can depend on it in the long term - but I think it's fine for now.

Reply:
> This isn't a stable name - so I'm not sure if we can depend on it in the long term - but I think it's fine for now.
@jgkamat 

Yes, this isn't ideal. The `ObjectIntrospection::getObjectSizeImpl` doesn't appear in the DWARF due to not having a body. I kind of think this is a bug, as the types are perfectly available, but it's a bug/feature that's largely out of our control. I guess the idea is that the strong symbol's debug would collide with the weak symbol's, and be more informative?

> Do we hardcode mangled names elsewhere?
@ot 

Not commonly, though there is one instance in [OICompiler.cpp:341](https://github.com/facebookexperimental/object-introspection/blob/23026d80ba9ac85409a3959f010eca4788737f0a/src/OICompiler.cpp#L341-L342).

The issue I'm concerned about with demangling this name is I'd then have to remangle it to get from the top one to the impl. If the impl was in the DWARF I could find it from there, but it isn't. Perhaps it would be more generic if I did `s/13getObjectSize/17getObjectSizeImpl/` but that also feels sketchy. Open to suggestions!

Reply:
Since writing that I've changed my mind, going to attempt to work with the demangled names and keep a reverse mapping.